colcon_core/shell/__init__.py

```diff
@@ -325,7 +326,13 @@ def get_shell_extensions():
                 "encoding '{encoding}': {line_replaced}".format_map(locals()))
             continue
         parts = line.split('=', 1)
-        if len(parts) == 2 and re.match('^[a-zA-Z0-9_%]+$', parts[0]):
+        if sys.platform != 'win32':
+            regex = '^[a-zA-Z_][a-zA-Z0-9_]*$'
+        else:
+            regex = '^[a-zA-Z0-9%' + ''.join(
+                '\\' + c for c in r'_(){}[]$*+-\/"#\',;.@!?'
```

Comment:
Does the underscore need to be escaped or can it be moved to the first part of the expression?

Reply:
I doesn't need to be escaped: 4333f6e00cbe35f716c27fd98e4aa4d985e83140.